### PR TITLE
`UnitUtils#getDimensionName` introduced

### PR DESCRIPTION
**Pull request by msteigenberger**
This makes it possible to dynamically get the `itemType` from value + unit which is sent from a device and therefore a channel-type can be dynamically created.
This is needed in PR https://github.com/openhab/openhab2-addons/pull/2167

### CODE REVIEW

File: bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/types/util/UnitUtils.java [hunk 1]
```diff
@@ -82,6 +92,39 @@
...
+            for (Field field : system.getDeclaredFields()) {
+                if (field.getType().isAssignableFrom(Unit.class) && Modifier.isStatic(field.getModifiers())) {
+                    try {
+                        String dimension = ((Class<?>) ((ParameterizedType) field.getGenericType())
+                                .getActualTypeArguments()[0]).getSimpleName();
+                        Unit<?> systemUnit = (Unit<?>) field.get(null);
+                        if (systemUnit.isCompatible(unit)) {
+                            return dimension;
+                        }
+                    } catch (Exception e) {
```
Comment (htreu) [inline bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/types/util/UnitUtils.java]: Are you able to catch something more specific then `Exception`?

Comment (msteigenberger) [reply bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/types/util/UnitUtils.java]: There are many exceptions that can be thrown here that are defined in the used java methods. Catching all of them would be superfluous imho as nothing can be done anyhow apart from logging.
Though there are some cases which I can handle specifically (but also just log them with a specific log message):
- Unit constant has no generic type argument
- The Unit value is null

Comment (htreu) [reply bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/types/util/UnitUtils.java]: Yes, please try to catch the specific exceptions. It's up to you wether you also want to do specific log entries, I would prefer it though.

Comment (msteigenberger) [reply bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/types/util/UnitUtils.java]: Ok now?

**Pull request merged**